contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -102,33 +102,155 @@ contract SellerHandlerFacet is SellerBase {
+              delete lookups.sellerIdByClerk[seller.clerk];
+              lookups.sellerIdByClerk[sender] = _sellerId;
+            }
+
+            // Delete pending update clerk
+            sellerPendingUpdate.clerk = address(0);
+
+        }
+
+        // Notify watchers of state change
+        emit SellerUpdated(_sellerId, seller, 0, sender);
+    }
+
+    function preSellerUpdateChecks(uint256 _sellerId, SellerUpdate calldata _seller) internal {
+        // Check that the operator passed in addresses are unique to one seller id across all roles -- not used or are used by this seller id.
+        if (_seller.operator != address(0)) {
             uint256 check1 = lookups.sellerIdByOperator[_seller.operator];
-            uint256 check2 = lookups.sellerIdByOperator[_seller.clerk];
+            uint256 check2 = lookups.sellerIdByClerk[_seller.operator];
             uint256 check3 = lookups.sellerIdByAdmin[_seller.operator];
-            uint256 check4 = lookups.sellerIdByAdmin[_seller.clerk];
-            uint256 check5 = lookups.sellerIdByClerk[_seller.operator];
-            uint256 check6 = lookups.sellerIdByClerk[_seller.clerk];
-            require(
+
+             require(
                 (check1 == 0 || check1 == _seller.id) &&
                     (check2 == 0 || check2 == _seller.id) &&
-                    (check3 == 0 || check3 == _seller.id) &&
-                    (check4 == 0 || check4 == _seller.id) &&
-                    (check5 == 0 || check5 == _seller.id) &&
-                    (check6 == 0 || check6 == _seller.id),
+                    (check3 == 0 || check3 == _seller.id),
                 SELLER_ADDRESS_MUST_BE_UNIQUE
             );
         }
 
-        // Admin address or AuthToken data must be present in parameters. A seller can have one or the other. Check passed in parameters
-        if (_seller.admin == address(0)) {
-            // Check that auth token is unique to this seller
-            uint256 check = lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId];
-            require(check == 0 || check == _seller.id, AUTH_TOKEN_MUST_BE_UNIQUE);
-        } else {
-            // Check that the admin address is unique to one seller id across all roles -- not used or is used by this seller id.
+        // Check that the clerk passed in addresses are unique to one seller id across all roles -- not used or are used by this seller id.
+        if (_seller.clerk != address(0)) {
+            uint256 check1 = lookups.sellerIdByOperator[_seller.clerk];
+            uint256 check2 = lookups.sellerIdByClerk[_seller.clerk];
+            uint256 check3 = lookups.sellerIdByAdmin[_seller.clerk];
+
+             require(
+                (check1 == 0 || check1 == _seller.id) &&
+                    (check2 == 0 || check2 == _seller.id) &&
+                    (check3 == 0 || check3 == _seller.id),
+                SELLER_ADDRESS_MUST_BE_UNIQUE
+            );
+        }
```

Comment:
Could this not be written as something like : 
```
uint256 check = 0;
if (_seller.clerk != address(0)) {
    check += lookups.sellerIdByOperator[_seller.clerk];
    check += lookups.sellerIdByClerk[_seller.clerk];
    check += lookups.sellerIdByAdmin[_seller.clerk];
}
require(check == 0, SELLER_ADDRESS_MUST_BE_UNIQUE);
```
Seems easier to read, i find the require statement rather hard to process / validate.

Reply:
can't, because we have to compare each check against the sellerId  ```check1 == _sellerId```, ```check2 == _sellerId```, etc.